File: backend/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, field_validator


class DocumentSubmit(BaseModel):
    extracted_data: dict
    confidence_scores: dict | None = None


class DocumentUpdate(BaseModel):
    extracted_data: dict | None = None
    confidence_scores: dict | None = None
    status: str | None = None
    comment: str | None = None


class AuditEventResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    document_id: str
    action: str
    actor: str
    field_name: str | None
    old_value: dict | None
    new_value: dict | None
    comment: str | None
    timestamp: datetime


class DocumentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    project_id: str
    document_type_id: str
    status: str
    extracted_data: dict
    confidence_score: float | None
    confidence_scores: dict | None
    created_at: datetime
    updated_at: datetime
    history: list[AuditEventResponse] | None = None


class DocumentListResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    project_id: str
    document_type_id: str
    status: str
    extracted_data: dict
    confidence_score: float | None
    confidence_scores: dict | None
    created_at: datetime
    updated_at: datetime
    document_type_name: str | None = None

File: frontend/src/types/document.ts
export interface AuditEvent {
  id: string;
  document_id: string;
  action: string;
  actor: string;
  field_name: string | null;
  old_value: Record<string, unknown> | null;
  new_value: Record<string, unknown> | null;
  comment: string | null;
  timestamp: string;
}

export interface Document {
  id: string;
  project_id: string;
  document_type_id: string;
  status: string;
  extracted_data: Record<string, unknown>;
  confidence_score: number | null;
  confidence_scores?: Record<string, number> | null;
  created_at: string;
  updated_at: string;
  history?: AuditEvent[] | null;
  document_type_name?: string | null;
}
